### PR TITLE
Added Traverse and Bitraverse instances to the CommittableConsumerRecord, CommittableProducerRecords, ConsumerRecord and ProducerRecord

### PR DESCRIPTION
**Pull request by LMnet**
In this pull request, I added `Traverse` and `Bitraverse` instances to the `CommittableConsumerRecord`, `CommittableProducerRecords`, `ConsumerRecord` and `ProducerRecord`. I was really missing these instances in everyday work.

Unfortunately, there are a few types that could benefit from these instances too, but they implementation does not give an opportunity to implement these instances. I'm talking about `ProducerResult`, `ProducerRecords`, and `TransactionalProducerRecords`. They all have a `P` (passthrough) type as the rightmost type. Scala's partial unification is right-biased. And because of this, we can only implement `Traverse` or `Bitraverse` for the rightmost types. Which would be inconsistent with all other types instances. I [created an issue](https://github.com/fd4s/fs2-kafka/issues/410) for moving `P` parameter to the left.

### CODE REVIEW

**Comment by LMnet** [comment]
The build is red because of this: https://github.com/fd4s/fs2-kafka/pull/406

**Comment by vlovgr** [comment]
Could you add the tests for checking the laws for these instances @LMnet?

**Comment by LMnet** [comment]
> Could you add the tests for checking the laws for these instances @LMnet?

Ok, I will add them.

**Comment by LMnet** [comment]
@vlovgr I added tests in the second commit.

For laws testing, I added cats-testkit-scalatest. And after that I found, that there are `CatsEquality.scala` and `CatsSuite.scala` files in the project. Looks like it was an old copy-paste before cats-testkit-scalatest was published as a separate artifact. So I deleted these copy-pasted files.

To properly test `Traverse` and `Bitraverse` instances I added `Eq` instances to a few types. All these new instances are covered with laws tests too.

Full list of changes of the second commit:
- Added Eq instances for:
  - Header
  - Headers
  - Timestamp
  - ConsumerRecord
  - ProducerRecord
  - CommittableOffset
  - CommittableConsumerRecord
  - CommittableProducerRecords
- Deleted CatsEquality.scala and CatsSuite.scala in favor of cats-testkit-scalatest.
- Added laws tests for all newly added Eq instances and
- Added laws tests for Traverse/Bitraverse instances for:
  - CommittableConsumerRecord
  - CommittableProducerRecords
  - ConsumerRecord
  - ProducerRecord

File: modules/core/src/main/scala/fs2/kafka/CommittableOffset.scala
Comment (LMnet) [inline modules/core/src/main/scala/fs2/kafka/CommittableOffset.scala:140]: I'm not sure that this is 100% accurate because `CommittableOffset` also has a `commit: F[Unit]`. In my implementation, I just ignore this field.

File: modules/core/src/main/scala/fs2/kafka/Header.scala
Comment (LMnet) [inline modules/core/src/main/scala/fs2/kafka/Header.scala:73]: Since there is no `Eq[Array[Byte]]` I converted arrays to a vector.

File: modules/core/src/test/scala/fs2/kafka/BaseGenerators.scala
Comment (LMnet) [inline modules/core/src/test/scala/fs2/kafka/BaseGenerators.scala:80]: Here I ignore `commit: F[Unit]` too, like in the `Eq` instance.

File: modules/core/src/main/scala/fs2/kafka/Header.scala
Comment (LMnet) [reply modules/core/src/main/scala/fs2/kafka/Header.scala]: I thought that converting to vector will have a too big performance impact. Now I'm using `sameElements`.

File: modules/core/src/main/scala/fs2/kafka/CommittableOffset.scala
Comment (bplommer) [reply modules/core/src/main/scala/fs2/kafka/CommittableOffset.scala:140]: Yeah, that makes me slightly nervous. Do you have any thoughts on this @vlovgr?

Comment (vlovgr) [reply modules/core/src/main/scala/fs2/kafka/CommittableOffset.scala:140]: I think this is the reason why these instances haven't been added before. In practice though, we should never have two consumers with the same (group-id, topic-partition, offset), so not sure how this would be an issue. 🤷‍♂️

Comment (LMnet) [reply modules/core/src/main/scala/fs2/kafka/CommittableOffset.scala:140]: I think, in practice, this will not be a problem. 

But in terms of API, it feels a bit weird. `CommittableOffset` feels like pure data, but it contains `def commit: F[Unit]`, which is not data, but a behavior. Maybe it would be better if `CommittableOffset` will not have `commit` method? Maybe it would be more natural to have it on `KafkaConsumer` (like `def commit(committableOffset: CommitableOffset): F[Unit]`)?

Of course, this change will not be backward compatible, and I think it not worth it. Maybe, someday, in some major update :)

Comment (vlovgr) [reply modules/core/src/main/scala/fs2/kafka/CommittableOffset.scala:140]: Would you mind opening an issue about this and we can schedule it for 2.0? :)

Comment (LMnet) [reply modules/core/src/main/scala/fs2/kafka/CommittableOffset.scala:140]: Ok: https://github.com/fd4s/fs2-kafka/issues/462

**Review by bplommer — APPROVED** [review]
LGTM - amazing work!

**Comment by bplommer** [comment]
Can we merge this @vlovgr?

**Comment by bplommer** [comment]
Looks like we're now blocked on a cats-testkit-scalatest release for 3.0.0-M3. I've opened a PR at https://github.com/typelevel/cats-testkit-scalatest/pull/109

**Review by vlovgr — APPROVED** [review]

**Comment by bplommer** [comment]
We might want to inline the dependency on `cats-testkit-scalatest` since it doesn't seem to be getting very prompt updates.

**Comment by LMnet** [comment]
> We might want to inline the dependency on cats-testkit-scalatest since it doesn't seem to be getting very prompt updates.

It would be sad if you have to do it. Maybe we could find some of the maintainers in gitter?

**Comment by bplommer** [comment]
> > We might want to inline the dependency on cats-testkit-scalatest since it doesn't seem to be getting very prompt updates.
> 
> 
> 
> It would be sad if you have to do it. Maybe we could find some of the maintainers in gitter?

I asked on `cats-dev` a few days ago but didn't get a reply. This will be a blocker every time there's a new Scala 3 milestone, so I'm reluctant to add a new dependency that  creates extra maintenance overhead unless there's a really strong reason.

**Comment by LMnet** [comment]
Well, it's reasonable.

At least let's add a TODO comment on this inlined stuff to remove it in the future.

**Comment by bplommer** [comment]
There's now a cat-testkit-scalatest release, so let's just use that.

There are some compile errors with Scala 3 - I can fix them later today.

**Comment by bplommer** [comment]
Thanks @LMnet - sorry for taking so long over this!

**Pull request merged**
